From a3c11d276d7e9e9991441816c25451bd699e96db Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@rvc.renesas.com>
Date: Thu, 2 Jun 2016 10:11:14 +0700
Subject: [PATCH 732/909] thermal: renesas: Add some property descriptions for
 emergency shutdown node

commit 70ddb0a1e168fbb1751618390b586d605d17a756 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch adds description for "status", "target_cpus" property
of emergency shutdown node and correct some typos.
Also, it modifies DT example for interrupt mode due to updated
IRQ implementation.

Signed-off-by: Hien Dang <hien.dang.eb@rvc.renesas.com>
(cherry picked from horms/renesas-bsp commit 957b800af000df9cf99ffeca7e03dd62ed453032)
[jiada: Resolve conflict in Documentation/devicetree/bindings/thermal/rcar-gen3-thermal.txt]
Signed-off-by: Jiada Wang <jiada_wang@mentor.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/thermal/rcar-gen3-thermal.txt     | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/thermal/rcar-gen3-thermal.txt b/Documentation/devicetree/bindings/thermal/rcar-gen3-thermal.txt
index 5d0ee4267f86..261c3fedcf29 100644
--- a/Documentation/devicetree/bindings/thermal/rcar-gen3-thermal.txt
+++ b/Documentation/devicetree/bindings/thermal/rcar-gen3-thermal.txt
@@ -57,27 +57,37 @@ Example:
 	};
 
 * Emergency shutdown for R-CAR Gen3
-Emergency shutdown functionality provide the specific cooling mechanism
+Emergency shutdown functionality provides the specific cooling mechanism
 for R-CAR Gen3. In case of high temperature(e.g over 100 degrees),
-it has the ability to rapidly cooling the system.
+it has the ability to rapidly cool down the system.
 
 Required property:
 - polling-delay:	The maximum number of milliseconds to wait between polls
   Type: unsigned	when checking temperature for emergency shutdown.
   Size: one cell
 
-- on-temperature:       This value indicate the emergency temperature and invoke
+- on-temperature:       This value indicates the emergency temperature and invokes
   Type: unsigned	emergency shutdown functionality when exceeding this
   Size: one cell	temperature.
 
-- off-temperature:	This value indicate the temperature to disable emergency
+- off-temperature:	This value indicates the temperature to disable emergency
   Type: unsigned	shutdown.
   Size: one cell
 
+- status:		Should be "disabled" always.
+  Type: string
+
+- target_cpus:		This property indicates which CPU will be targeted for shutdown.
+  Type: phandle
+
 thermal-zones {
 	emergency {
 		polling-delay = <1000>;    /* milliseconds */
 		on-temperature = <110000>; /* millicelsius */
 		off-temperature = <95000>; /* millicelsius */
+		target_cpus = <&a57_1>,
+			      <&a57_2>,
+			      <&a57_3>;
+		status = "disabled";
 	};
 };
-- 
2.17.1

